### PR TITLE
[datadog_logs_custom_pipeline] Add reference table logs processor

### LINKED ISSUE #1589: Cannot import datadog_logs_custom_pipeline containing a lookup_processor
### Terraform Version

```
terraform -v
Terraform v1.0.1
on darwin_amd64
+ provider registry.terraform.io/datadog/datadog v3.15.1
```

(Confirmed on almost all 3.x versions too)

### Affected Resource(s)

-   datadog_logs_custom_pipeline

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    datadog = {
      source = "DataDog/datadog"
    }
  }
}

provider "datadog" {
  api_url = "https://api.datadoghq.eu/"
}

resource "datadog_logs_custom_pipeline" "organization_lookup" {
  name       = "Organization Lookup"
  is_enabled = true

  filter {
    query = ""
  }

  processor {
    lookup_processor {
      name           = "Organization ID-Detail lookup"
      is_enabled     = true

      source         = "organization_id"
      target         = "organization_detail"

      lookup_table   = [
        "1,google",
        "2,microsoft",
      ]
    }
  }
}
```

### Debug Output

https://gist.github.com/jeromepin/5ef1db7096c0c5b3d4fc54590ef610aa

### Panic Output

None

### Expected Behavior

Either :
- ignore the `lookup_enrichment_table` attribute when importing. It will create drift but since the attribute isn't supported in TF, I believe it's up to the user to work around.
- support for the `lookup_enrichment_table` attribute (Relates to #1098)

### Actual Behavior

```
Error: object contains unparsed element: map[is_enabled:true lookup_enrichment_table:organization name:Organization ID-Detail lookup source:organization_id target:organization_detail type:lookup-processor]
```

### Steps to Reproduce

1. `terraform import 'datadog_logs_custom_pipeline.organization_lookup' '$ID'`

### Important Factoids

The pipeline has a lookup processor using an enrichment table instead of manually-provided table.

### References

-   #1098

### PR DESCRIPTION
**Pull request by nkzou**
SSIA. Resolves #1589 and #1098

### CODE REVIEW

**Review by rtrieu — APPROVED** [review]
LGTM

File: datadog/tests/data_source_datadog_logs_pipelines_test.go [hunk 1]
```diff
@@ -16,7 +16,6 @@ import (
 )
 
 func TestAccDatadogLogsPipelinesDatasource(t *testing.T) {
-	t.Parallel()
```
Comment (skarimo) [inline datadog/tests/data_source_datadog_logs_pipelines_test.go:19]: Any reason this was removed?

**Review by skarimo — DISMISSED** [review]

File: docs/resources/logs_custom_pipeline.md [hunk 2]
```diff
@@ -388,6 +389,7 @@ Optional:
...
 - `lookup_processor` (Block List, Max: 1) Lookup Processor. More information can be found in the [official docs](https://docs.datadoghq.com/logs/processing/processors/?tab=ui#lookup-processor) (see [below for nested schema](#nestedblock--processor--pipeline--processor--lookup_processor))
 - `message_remapper` (Block List, Max: 1) Message Remapper Processor. More information can be found in the [official docs](https://docs.datadoghq.com/logs/processing/processors/?tab=ui#log-message-remapper) (see [below for nested schema](#nestedblock--processor--pipeline--processor--message_remapper))
+- `reference_table_lookup_processor` (Block List, Max: 1) Reference Table Lookup Processor. More information can be found in the [official docs](https://docs.datadoghq.com/logs/processing/processors/?tab=ui#lookup-processor) (see [below for nested schema](#nestedblock--processor--pipeline--processor--reference_table_lookup_processor))
```
Comment (vinckama) [inline docs/resources/logs_custom_pipeline.md:392]: I *think* it's useful to re-specify here that reference tables are in public beta.

```suggestion
- `reference_table_lookup_processor` (Block List, Max: 1) Reference Table Lookup Processor. Reference Tables are in public beta. More information can be found in the [official docs](https://docs.datadoghq.com/logs/processing/processors/?tab=ui#lookup-processor) (see [below for nested schema](#nestedblock--processor--pipeline--processor--reference_table_lookup_processor))
```

**Review by vinckama — APPROVED** [review]
approved with one comment

**Review by skarimo — APPROVED** [review]

**Pull request merged**